app/services/form_validator.rb

```diff
@@ -60,6 +61,20 @@ def validate_country(form_element, el_name, errors)
     end
   end
 
+  def validate_postal(form_element, el_name, errors)
+    zip = @params[el_name]
+    country = @params.fetch(:country, :US)
+    if country.respond_to? :to_sym
+      country = country.to_sym
+    else
+      # country is likely to be nil, so set it to our default
+      country = :US
+    end
+    if form_element.data_type == 'postal' && zip.present? && !is_zip(zip, country)
```

Comment:
it's a bit confusing that you refer to it as zip and postal at different places in the codebase